core/src/test/resources/sql/measure.iq

```diff
@@ -26,6 +26,136 @@ from emp;
 
 !update
 
+create view empm_comm as
+select *, avg(sal) as measure avg_sal, avg(comm) as measure avg_comm
+from emp;
+(0 rows modified)
+
+!update
+
+# AGGREGATE with GROUP BY of a column that includes NULLs
+select mgr, avg_comm from empm_comm
+group by mgr;
++------+----------+
+| MGR  | AVG_COMM |
++------+----------+
+| 7566 |          |
+| 7698 |   550.00 |
+| 7782 |          |
+| 7788 |          |
+| 7839 |          |
+| 7902 |          |
+|      |          |
++------+----------+
+(7 rows)
+
+!ok
+
+# AGGREGATE with GROUP BY of an expression
+SELECT (job || '_yo') as job_yo, avg_sal FROM empm
+GROUP BY (job || '_yo');
++--------------+---------+
+| JOB_YO       | AVG_SAL |
++--------------+---------+
+| ANALYST_yo   | 3000.00 |
+| CLERK_yo     | 1037.50 |
+| MANAGER_yo   | 2758.33 |
+| PRESIDENT_yo | 5000.00 |
+| SALESMAN_yo  | 1400.00 |
++--------------+---------+
+(5 rows)
+
+!ok
+
+# single GROUPING SETS group with AS MEASURE reference
+SELECT job, year(hiredate) as hire_year, avg_sal FROM empm
```

Comment:
if you're going to take 4 lines for GROUP BY, at least put FROM on its own line. (I find queries much more readable if clauses are on separate lines.)

Reply:
This file is ordered by complexity - simple queries first - and groups related concepts. `GROUPING SETS` is relatively advanced functionality, so I would move these queries further down the `measure.iq` file.